xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/metadata/SolrLinkSerializer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.search.solr.internal.metadata;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceResolver;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+
+/**
+ * Help serialize and unserialize links stored in Solr.
+ * 
+ * @version $Id$
+ * @since 14.8RC1
+ */
+@Component(roles = SolrLinkSerializer.class)
+@Singleton
+public class SolrLinkSerializer
```

Comment:
Instead of relying on this custom implem, wouldn't it worth it to have it implemented in an EntityReference <-> String converter, so that it could be directly use with SolrUtils to convert the field to the appropriate value?

Reply:
It's not an EntityReference <-> String converter, but more a SomeFutureLinkType <-> String converter.

Reply:
> here you just need to convert a String to an EntityReference with parameters no?

No. We have resolvers and serializes for that (and that's what SolrLinkSerializer, notice the "withparameters" role hints). The point of SolrLinkSerializer is to serialize a **link** and currently it only supports entity kind of links.